1 Holiday Rental Property Management Tool/submissions/enricop89/README.md

```diff
@@ -0,0 +1,21 @@
+# Holiday Rental Property Management Tool
+
+For this project I imagined to host the client side application in an S3 bucket backed by a Cloudfront distribution to enforce HTTPS, global distribution and reduced latency. The client side application will have two main users: guests and hosts. Hosts can login using AWS Cognito, Guests can only see the property and book a room.
+
+The Host flow is the following:
+
+- Host signs up for the platform and chooses his own subdomain
+- if available, a new record is created on S3 and a dedicated user pool in AWS Cognito
+- they connect their Stripe account
+
+The main CRUD operations are implemented through an API Gateway with a specific Lambda function for each operation. The Lambda functions stores record in a DynamoDB Table. The DynamoDB table has Dynamo Streams configured so everytime there is a new reservation, an update on an existing reservation or a cancelled reservation, the update is reflected into DynamoDB Streams and can be streamed to 3rd party platform to keep the calendar synched. The same behaviour, on the other way around, so when people book from 3rd party platform, triggers a webhook which is a Lambda function behind the same API gateway, which creates a record in DynamoDB.
+
+All the notifications through emails are handled by AWS SES, which will be called from Lambda functions when needed (new reservations, cleaners, doors key).
+
+Each of the automated tasks such as check in in 10 days, cleaners notifications, checkout schedulers, door keys are handled with an Amazon Eventbridge Scheduler which runs daily and query the dynamoDB table (by PK to keep it efficient) to retrieve reservations and send notifications accordingly. There will be a scheduler per each notification
```

Comment:
If you've got hundreds of thousands of bookings, how would you design the pk/ sk to do the query effectively?

Reply:
I explained pk/sk structure in the Q&A file: https://github.com/SamWSoftware/ServerlessArchitectureChallenge/pull/1/files#diff-c816a4126e2d4b66d60c9d7fe976fcb1b0b0a0262d72e7965add7706ff9af538R100